src/mscorlib/src/System/Guid.cs

```diff
@@ -365,214 +365,287 @@ public static bool TryParseExact(String input, String format, out Guid result)
             }
         }
 
-
-        private static bool TryParseGuid(String g, GuidStyles flags, ref GuidResult result)
+        private static bool TryParseGuid(string g, GuidStyles flags, ref GuidResult result)
         {
-            if (g == null) {
+            if (g == null)
+            {
                 result.SetFailure(ParseFailureKind.Format, "Format_GuidUnrecognized");
                 return false;
             }
-            String guidString = g.Trim();  //Remove Whitespace
+            
+            // Consume leading and trailing whitespace
+            int start = 0;
+            while (start < g.Length && char.IsWhiteSpace(g[start]))
+            {
+                start++;
+            }
 
-            if (guidString.Length == 0) {
+            int end = g.Length;
+            while (end > start && char.IsWhiteSpace(g[end - 1]))
+            {
+                end--;
+            }
+
+            int length = end - start;
+            if (length == 0)
+            {
+                // Empty string (or only whitespace)
                 result.SetFailure(ParseFailureKind.Format, "Format_GuidUnrecognized");
                 return false;
             }
 
-            // Check for dashes
-            bool dashesExistInString = (guidString.IndexOf('-', 0) >= 0);
-
-            if (dashesExistInString) {
-                if ((flags & (GuidStyles.AllowDashes | GuidStyles.RequireDashes)) == 0) {
-                    // dashes are not allowed
-                    result.SetFailure(ParseFailureKind.Format, "Format_GuidUnrecognized");
-                    return false;
-                }
-            }
-            else {
-                if ((flags & GuidStyles.RequireDashes) != 0) {
-                    // dashes are required
-                    result.SetFailure(ParseFailureKind.Format, "Format_GuidUnrecognized");
-                    return false;
+            bool hasDashesInString = false;
```

Comment:
NIT: It would be ideal if we didn't rename these variables.  The reason being that it will reduce some of the noise in the diff, which will be helpful not only for review but also help pushing changes back into the Desktop code base (where we like to see small diffs).